packages/backend-defaults/src/alpha/entrypoints/actionsRegistry/DefaultActionsRegistryService.ts

```diff
@@ -64,27 +66,65 @@ export class DefaultActionsRegistryService implements ActionsRegistryService {
     const router = PromiseRouter();
     router.use(json());
 
-    router.get('/.backstage/actions/v1/actions', (_, res) => {
+    const authorizeAction = async (
+      actionId: string,
+      credentials: BackstageCredentials,
+      input?: AnyZodObject,
+    ) => {
+      const action = this.actions.get(actionId);
+      if (!action) {
+        return false;
+      }
+      if (!action.authorize) {
+        return true;
+      }
+
+      try {
+        const decision = await action.authorize({ credentials, input });
+        return decision.result === AuthorizeResult.ALLOW;
+      } catch (err) {
+        this.logger.error(`Failed to authorize action ${actionId}`, err);
+        return false;
+      }
+    };
+
+    router.get('/.backstage/actions/v1/actions', async (req, res) => {
```

Comment:
Don't think this will always be the way that actions are accessed. I wonder if it might be best to have the permission be returned in the metadata instead? For example just having an `authorized: boolean` field in the response per action, leaving it to the caller to decide how to represent actions that the user doesn't have access to.